src/applications/mhv-landing-page/tests/containers/AppConfig.unit.spec.jsx

```diff
@@ -1,80 +1,76 @@
-import { datadogRum } from '@datadog/browser-rum';
 import { waitFor } from '@testing-library/react';
 import { renderWithStoreAndRouter } from '@department-of-veterans-affairs/platform-testing/react-testing-library-helpers';
+import {
+  setDatadogRumUser,
+  useDatadogRum,
+} from '@department-of-veterans-affairs/mhv/exports';
 import sinon from 'sinon';
 import { expect } from 'chai';
 
 import AppConfig from '../../containers/AppConfig';
 import { appName } from '../../manifest.json';
 
-const initialStateFn = ({ accountUuid = 'test-id' } = {}) => ({
-  user: {
-    profile: {
-      accountUuid,
+describe(`${appName} -- <AppConfig />`, () => {
+  const initialStateFn = ({ accountUuid = 'test-id' } = {}) => ({
+    user: {
+      profile: {
+        accountUuid,
+      },
     },
-  },
-});
+  });
 
-describe(`${appName} -- <AppConfig />`, () => {
   it('initializes datadog RUM', () => {
-    const useRumSpy = sinon.spy(datadogRum, 'init');
-
+    const useRumSpy = sinon.spy(useDatadogRum);
     const initialState = initialStateFn();
 
-    const { getByText } = renderWithStoreAndRouter(
-      <AppConfig>
+    const { findByText } = renderWithStoreAndRouter(
+      <AppConfig useDatadogRumFn={useRumSpy}>
         <p>child node</p>
       </AppConfig>,
       { initialState },
     );
 
-    waitFor(() => {
-      getByText('child node');
+    findByText('child node');
+    expect(useRumSpy.called).to.be.true;
+  });
 
-      // Check that datadogRum methods are called
-      expect(useRumSpy.called).to.be.true;
+  describe('calls datadogRum.setUser', () => {
+    let setRumUserSpy;
+    beforeEach(() => {
+      setRumUserSpy = sinon.spy(setDatadogRumUser);
     });
 
-    useRumSpy.restore();
-  });
+    it('when a user id is available', async () => {
+      const initialState = initialStateFn();
 
-  it('calls datadogRum.setUser when a user id is available', () => {
-    const setRumUserSpy = sinon.spy(datadogRum, 'setUser');
-    const initialState = initialStateFn();
-
-    const { getByText } = renderWithStoreAndRouter(
-      <AppConfig>
-        <p>child node</p>
-      </AppConfig>,
-      { initialState },
-    );
+      const { getByText } = renderWithStoreAndRouter(
+        <AppConfig setDatadogRumUserFn={setRumUserSpy}>
+          <p>child node</p>
+        </AppConfig>,
+        { initialState },
+      );
 
-    waitFor(() => {
-      getByText('child node');
-      // Check that datadogRum methods are called
-      expect(setRumUserSpy.calledWith({ id: 'test-id' })).to.be.true;
+      await waitFor(() => {
+        getByText('child node');
+        expect(setRumUserSpy.calledWith({ id: 'test-id' })).to.be.true;
+      });
     });
 
-    setRumUserSpy.restore();
-  });
-
-  it('does not call datadogRum.setUser when user id is not available', () => {
-    const setRumUserSpy = sinon.spy(datadogRum, 'setUser');
-    const initialState = initialStateFn({ accountUuid: '' });
+    it('when user id is not available', async () => {
```

Comment:
The test description now says the opposite of what it should: "calls datadogRum.setUser" + "when user id is not available", which is incorrect. It does/should _not_ call that method when a user is not available.

Perhaps this:

```suggestion
    it('unless user id is not available', async () => {
```

Reply:
Good find! Fixed.